360/watchapp/views.py

```diff
@@ -51,22 +192,19 @@ def register_view(request):
             error = "Passwords do not match."
         else:
             # Create a new user
-            user = User.objects.create_user(
-                username=username, password=password, 
-                first_name=first_name, last_name=last_name)
+            user = User.objects.create(
+                username=username,
+                password=make_password(password),
+                first_name=fullname
+            )
             messages.success(request, 'Registration successful! Redirecting to login...')
             return redirect('login')
 
     return render(request, 'watchapp/register.html', {'error': error})
 
-def homepage_view(request):
-    # Get all friends of the logged-in user
-    friends = Friend.objects.friends(request.user)
-
-    # Create a list of usernames from the friends queryset
-    friends_data = [{'username': friend.username} for friend in friends]
-
-    return render(request, 'watchapp/homepage.html', {"friends": friends_data})
+def homepage(request):
```

Comment:
I might be wrong, but I'm not sure where this view is being called. I don't see it called in urls.py, only homepage_view is called. Just something to look into.

Reply:
Deleted. It was old code.